core/src/test/scala/unit/kafka/integration/MetricsDuringTopicCreationDeletionTest.scala

```diff
@@ -0,0 +1,149 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.integration
+
+import java.util.Properties
+import kafka.server.KafkaConfig
+import kafka.utils._
+import util.control.Breaks._
+import scala.collection.JavaConverters.mapAsScalaMapConverter
+
+import org.apache.log4j.{Level, Logger}
+import org.junit.{After, Before, Test}
+import com.yammer.metrics.Metrics
+import com.yammer.metrics.core.Gauge
+
+
+class MetricsDuringTopicCreationDeletionTest extends KafkaServerTestHarness with Logging {
+  val log = Logger.getLogger(classOf[MetricsDuringTopicCreationDeletionTest])
+  val numNodes = 3
+  val topic = "topic"
+  val topicNum = 2
+  val replicationFactor = 3
+  val partitionNum = 1
+  val createDeleteIterations = 3
+  
+  val overridingProps = new Properties
+  overridingProps.put(KafkaConfig.DeleteTopicEnableProp, "true")
+  overridingProps.put(KafkaConfig.AutoCreateTopicsEnableProp, "false")
+
+  val myMetrics = List("OfflinePartitionsCount","PreferredReplicaImbalanceCount", "UnderReplicatedPartitions")
+  val myTopics = List.tabulate(topicNum) (n => topic + n)
+
+  override def generateConfigs() = TestUtils.createBrokerConfigs(numNodes, zkConnect)
+    .map(KafkaConfig.fromProps(_, overridingProps))
+
+  @Before
+  override def setUp {
+    // prior harness runs may have left a populated registry
+    // each instance of a controller apparently creates new metrics but the 'newGauge'
+    // method actually invokes 'MetricsRegistry.getOrAdd' 
+    // so with multiple controllers in the same JVM only the first controller actually creates metrics.
+    for (m <- myMetrics) {
+        Metrics.defaultRegistry.allMetrics.asScala
+        .filterKeys(k => k.getName.endsWith(m))
+        .headOption match {
+           case Some(e) => Metrics.defaultRegistry.removeMetric(e._1)
+           case None =>
+        }
+    }
+    
+    super.setUp
+    log.setLevel(Level.INFO)
+  }
+
+  @After
+  override def tearDown {
+    super.tearDown
+  }
+
+  /*
+   * checking all metrics we care in a single test is faster 
+   */
+  @Test
+  def testMetricsDuringTopicCreateDelete() {
+    for (m <- myMetrics) {
+      testMetricDuringCreateDelete(m)
+    }
+  }
+
+  def testMetricDuringCreateDelete(metricName: String) {
+    val gauge = Metrics.defaultRegistry.allMetrics.asScala
+           .filterKeys(k => k.getName.endsWith(metricName))
+           .headOption
+           .getOrElse { fail( "Unable to find metric " + metricName ) }
+           ._2.asInstanceOf[Gauge[Int]]
+
+    var gaugeValue = gauge.value
+    assert(gaugeValue == 0)
+    
+    // Thread checking the metric continuously
+    var running = true
```

Comment:
We'll add it, it's good coding, though on both Linux and Mac we had the test working anyway.